agent/discover/sapsystem.go

```diff
@@ -0,0 +1,51 @@
+package discover
+
+import (
+	"github.com/trento-project/trento/internal/consul"
+	"github.com/trento-project/trento/internal/sapsystem"
+)
+
+const SAPDiscoverId string = "discover_sap"
+
+type SAPSystemsDiscover struct {
+	Id         string
+	host       Discover
+	SAPSystems sapsystem.SAPSystemsList
+}
+
+func (s SAPSystemsDiscover) GetId() string {
+	return s.Id
+}
+
+func (s SAPSystemsDiscover) ShouldDiscover(client consul.Client) bool {
+	return true
+}
+
+func (s SAPSystemsDiscover) storeDiscovery(cStorePath, cStoreValue string) error {
+	return nil
+}
+
+func (discover SAPSystemsDiscover) Discover() error {
+	systems, err := sapsystem.NewSAPSystemsList()
+
+	if err != nil {
+		return err
+	}
+
+	discover.SAPSystems = systems
+	for _, s := range discover.SAPSystems {
+		err := s.Store(discover.host.client)
```

Comment:
I'm a bit confused by this method. Aren't we supposed to use `storeDiscovery`? Why is that a no-op and we are calling another method on the SAPSystem struct instead?

Maybe we should drop `storeDiscovery` from the interface altogether?

Reply:
Yes. As my code does the KV storing outside the discovery I don't need to use it.

Reply:
Ok. I'm opening a follow-up issue on this.

Reply:
The idea of using storeDiscovery was that all implementations use a shared prefix on where they store the artifacts. that means we do not need to deduplicate key-values (e.g. if two implementations try to store "id", they can do so without overwriting each other). 

why are we deviating from that here?

Reply:
Because it is much easier to store whole `struct`s, that will have this `prefix` in their code. Keep in mind that the SAP system has dozens of values that we need to store. It is not a good idea to store them individually. That's why we create the struct with the whole obtained data, we convert it to map, and we store it. And this is done in the `Store` method for each of the individuals fields (this is the SAPSystem, but we would have the same for Cluster).  
This way, we abstract the KV usage in the discovery loop, this is now done by the new code.